src/persist-client/src/internal/encoding.rs

```diff
@@ -642,20 +643,49 @@ impl<T: Timestamp + Codec64> RustType<ProtoHollowBatch> for HollowBatch<T> {
     fn into_proto(&self) -> ProtoHollowBatch {
         ProtoHollowBatch {
             desc: Some(self.desc.into_proto()),
-            keys: self.keys.into_proto(),
+            parts: self.parts.into_proto(),
             len: self.len.into_proto(),
+            deprecated_keys: vec![],
         }
     }
 
     fn from_proto(proto: ProtoHollowBatch) -> Result<Self, TryFromProtoError> {
+        let mut parts: Vec<HollowBatchPart> = proto.parts.into_rust()?;
+        // MIGRATION: We used to just have the keys instead of a more structured
+        // part.
+        parts.extend(
+            proto
+                .deprecated_keys
+                .into_iter()
+                .map(|key| HollowBatchPart {
+                    key: PartialBatchKey(key),
+                    encoded_size_bytes: 0,
```

Comment:
could instead make it an Option, so it's more obvious when handling it in bounded mem compaction?